jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/ExtractTarStep.java

```diff
@@ -77,83 +82,116 @@ static boolean isGzipped(Path path) throws IOException {
   }
 
   private final Path tarPath;
-  private final Path destination;
+  private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
+  private final BuildConfiguration buildConfiguration;
 
-  ExtractTarStep(Path tarPath, Path destination) {
+  ExtractTarStep(
+      Path tarPath,
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
+      BuildConfiguration buildConfiguration) {
     this.tarPath = tarPath;
-    this.destination = destination;
+    this.progressEventDispatcherFactory = progressEventDispatcherFactory;
+    this.buildConfiguration = buildConfiguration;
   }
 
   @Override
   public LocalImage call()
       throws IOException, LayerCountMismatchException, BadContainerConfigurationFormatException {
-    Files.createDirectories(destination);
-    FileOperations.deleteRecursiveOnExit(destination);
-    TarExtractor.extract(tarPath, destination);
-
-    InputStream manifestStream = Files.newInputStream(destination.resolve("manifest.json"));
-    DockerManifestEntryTemplate loadManifest =
-        new ObjectMapper()
-            .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
-            .readValue(manifestStream, DockerManifestEntryTemplate[].class)[0];
-    manifestStream.close();
-    ContainerConfigurationTemplate configurationTemplate =
-        JsonTemplateMapper.readJsonFromFile(
-            destination.resolve(loadManifest.getConfig()), ContainerConfigurationTemplate.class);
-
-    List<String> layerFiles = loadManifest.getLayerFiles();
-    if (configurationTemplate.getLayerCount() != layerFiles.size()) {
-      throw new LayerCountMismatchException(
-          "Invalid base image format: manifest contains "
-              + layerFiles.size()
-              + " layers, but container configuration contains "
-              + configurationTemplate.getLayerCount()
-              + " layers");
-    }
+    Path destination = Files.createTempDirectory("jib-extract-tar");
+    try (TimerEventDispatcher ignored =
+        new TimerEventDispatcher(
+            buildConfiguration.getEventHandlers(),
+            "Extracting tar " + tarPath + " into " + destination)) {
+      Files.createDirectories(destination);
```

Comment:
nit: this probably won't need anymore. Maybe call `createTempDirectory()` here to make it close to `deleteRecursiveOnExit()`.

Reply:
The destination is used in the timer event string, so I can't move the creation inside, but yeah I can remove this line.